src/main/scala/is/hail/utils/Py4jUtils.scala

```diff
@@ -27,14 +27,13 @@ trait Py4jUtils {
     list
   }
 
-  // FIXME: Currently this can only send much less than the limit of
-  // BlockMatrix.toLocalMatrix. That's probably fine.
-  def doubleBDMToBytes(mat: BDM[Double]): Array[Byte] = {
-    // assumes matrix is written as written by BlockMatrix
-    val buf = new Array[Byte](8 + 8 * mat.rows * mat.cols)
-    Memory.storeInt(buf, 0, mat.rows)
-    Memory.storeInt(buf, 4, mat.cols)
-    Memory.memcpy(buf, 8, mat.data, 0, 8*mat.rows*mat.cols)
+  // FIXME: Currently this needs 8 * r * c < int.max.
```

Comment:
The factor of 8 is definitely going to rear it's head, it's the difference between 5792 and 46340 for side of square matrix.

Now that you have the assert, I think the FIXME should be made more specific about what to do. Send the data in smaller chunks, yeah?